### ISSUE TITLE
BigR[bigr.connect]: Required library 'BigRResultSet' could not be found

### ISSUE THREAD

**Issue by pregazzoni** [snowch/biginsight-examples#16]
Getting following error running BigR example. Where should lib path be set?

```
:Example
running
  '/Library/Frameworks/R.framework/Versions/3.2/Resources/bin/R --slave --no-restore --file=/Users/pierreregazzoni/biginsight-examples/examples/BigR/example.r'

Loading required package: rJava
Loading required package: methods
Error: BigR[bigr.connect]: Required library 'BigRResultSet' could not be found.
Execution halted
:Example FAILED
```

**Comment by snowch** [comment]
The lib dir is set in the build.gradle file:

```
environment 'libdir', "${projectDir}/lib"
```

https://github.com/snowch/biginsight-examples/blob/master/examples/BigR/build.gradle#L59
https://github.com/snowch/biginsight-examples/blob/master/examples/BigR/build.gradle#L73

Would you be able to test on a pristine virtual environment so that I can see if the issue is with the cluster or your client machine?
- install [virtualbox](https://www.virtualbox.org/) on your client machine
- install [vagrant](https://www.vagrantup.com/) on your client machine
- open a terminal
- ensure you are in the top level directory that contains Vagrantfile
- run `vagrant up`

**Comment by pregazzoni** [comment]
Yeah I saw libdir in build.gradle but somehow BigRResult jar is not getting picked up. Will try a clean report. Not familiar with vagrant but i assume should be straight forward.

**Comment by pregazzoni** [comment]
test on ubuntu (via vagrant) worked:

```
==> default: Running BigX tests
==> default: Test passed  ./examples/BigR
```

From output:

```
:Example running
  '/usr/lib/R/bin/R --slave --no-restore --file=/home/vagrant/biginsight-examples/examples/BigR/example.r'

Loading required package: rJava
Loading required package: methods

[1] TRUE
[1] ">> Connectivity test was successful."
```

So need to figure out what is diff on my MBP.

**Comment by zhaol2046** [comment]
@pregazzoni 
Hi pregazzoni, I have the same issue in my MBP. Do you have any idea how to fix that? Thanks!

**Comment by snowch** [comment]
@zhaol2046 can you do a `git pull` and try running:  

```
cd examples/BigR
./gradlew clean
./gradlew Connect -PdebugBigR 2>&1 > debug.output
```

Can you add a comment with `debug.output` attached after verifying your credentials aren't in the file?

**Comment by snowch** [comment]
@pregazzoni @zhaol2046  it's a long shot, but does adding this before `library(bigr)` in connect.r help:

```
jars <- list.files(pattern="\\.jar$", ignore.case=TRUE, recursive=TRUE, path="lib/")
for (j in jars) {
    .jaddClassPath(j)
}
```

ensure you do a `git pull` to get the latest updates

**Comment by zhaol2046** [comment]
@snowch I run the following commends in R and get the same error

```
> jars <- list.files(pattern="\\.jar$", ignore.case=TRUE, recursive=TRUE, path="lib/")
> for (j in jars) {
+     .jaddClassPath(j)
+ }
> library(bigr)
Loading required package: rJava
> bigr.connect(host="ehaasp-577-master-3.bi.services.bluemix.net", user="user10", password="temp4now", ssl = TRUE, trustStorePath = "/tmp/alex-truststore.jks", trustStorePassword = "mypassword", keyManager = "SunX509")
Error: BigR[bigr.connect]: Required library 'BigRResultSet' could not be found.
```

**Comment by zhaol2046** [comment]
@snowch The following is what I got after running ./gradlew clean and ./gradlew Connect -PdebugBigR 2>&1 > debug.output
`FAILURE: Build failed with an exception.
- What went wrong:
  Task 'Connect' not found in root project 'biginsight-examples'.
- Try:
  Run gradlew tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.`

I attach a screenshot of "debug.ouput" as follows...

<img width="1365" alt="screen shot 2016-06-21 at 4 29 03 pm" src="https://cloud.githubusercontent.com/assets/9555195/16245112/518bfc52-37cd-11e6-9f0e-40cd6c696f36.png">

**Comment by snowch** [comment]
@zhaol2046  You need to change into the `examples/BigR` directory before running `./gradlew ...`

**Comment by snowch** [comment]
See also: http://stackoverflow.com/questions/37954159/error-bigrbigr-connect-required-library-bigrresultset-could-not-be-found which I just posted.

**Comment by zhaol2046** [comment]
@snowch Thanks, waiting for solutions..

**Comment by snowch** [comment]
I've created a new target which will create a complete r script for you:

```
cd examples/BigR
./gradlew GenerateSampleScript
```

Then open the sample script `connect_template.r' in R or RStudio and run it manually.  You can run each statement line by line and verify that it completes successfully.

**Comment by zhaol2046** [comment]
@snowch Thanks for your efforts Chris. All the lines work fine except the bigr.connect...still get the same error...
`> connected <- bigr.connect(
    host = hostname,
     user = username,
     password = password,
     ssl = TRUE,
     trustStorePath = paste(projdir, "/truststore.jks", sep=""),
     trustStorePassword = "mypassword",
     keyManager = "SunX509"
 )
Error: BigR[bigr.connect]: Required library 'BigRResultSet' could not be found.`

**Comment by pregazzoni** [comment]
BTW testing with new IOP version and don't see this issue anymore on my MBP ...

```
:Connect
running
  '/Library/Frameworks/R.framework/Versions/3.2/Resources/bin/R --slave --no-restore --vanilla --file=/Users/pierreregazzoni/biginsight-examples/examples/BigR/connect.r'

Loading required package: rJava
Loading required package: methods
[1] "Successfully connected to BigR"
:KMeans
running
```

**Comment by snowch** [comment]
Ah, interesting.  Thanks @pregazzoni

**Comment by snowch** [comment]
@zhaol2046  you could try the virtualmachine that is included with this project.  The steps to try it are as follows:  
- install [virtualbox](https://www.virtualbox.org/)
- install [vagrant](https://www.vagrantup.com/)
- open a terminal 
- ensure you are in the top level directory that contains Vagrantfile
- run `vagrant up` on your client machine
- run `vagrant ssh -c "hostname -I | cut -d' ' -f2-"`
- open a web browser using the IP address from the previous step and the port 8787 - http://youripaddress:8787
- login using username: vagrant, password: vagrant
- in rstudio file navigator, open `biginsight-examples/examples/BigR/connect_template.r`
- execute this file in rstudio to connect to your cluster

cc @pregazzoni - rsudio server added to the vagrant project ^
